### PR TITLE
add an env--app dropdown to the services dashboard

### PR DESCRIPTION
**Pull request by azylman**
To make it easier to use. This is pretty hacky because grafana isn't really designed for this, the right way to do it would be to write a custom data source. But... I didn't want to because it was more work.

### CODE REVIEW

File: dashboards/scripted_services.js [hunk 1]
```diff
@@ -79,38 +110,65 @@ return function(callback) {
...
   };
 
-  if (_.isUndefined(ARGS.env) || _.isUndefined(ARGS.app)) {
-    dashboard.rows.push(
-        getErrorRow("## Missing env and/or app in ARGS")
-    );
-    return callback(dashboard);
-  }
+  init(function() {
+    // Get the ECS cluster, env, and appID
```
Comment (peggyl) [inline dashboards/scripted_services.js]: Remove `cluster`

File: dashboards/scripted_services.js [hunk 2]
```diff
@@ -62,8 +64,37 @@ var window, document, ARGS, $, jQuery, moment, kbn;
 
 var config = {
   ecsMetricsURL: "https://production--ecs-metrics.int.clever.com",
-  defaultDevCluster: "ecs-dev-Cluster-1EO83BILGWJL8",
-  defaultProdCluster: "ecs-production-Cluster-7M1CTQ6EN27T"
+}
+
+function init(callback) {
```
Comment (peggyl) [inline dashboards/scripted_services.js]: Briefly document what's happening here so anyone else working on this in the future has some starting knowledge.

**Comment by dfuentes** [comment]
I mean, code-wise this looks good to me.  It's a bit janky, and I'm slightly worried about the repeated service lookup, but it is a better user experience still.

File: dashboards/scripted_services.js [hunk 3]
```diff
@@ -79,38 +110,65 @@ return function(callback) {
...
-      dashboard.rows.push(
-        getErrorRow("Could not find matching service for " + env + "/" + app)
-      );
-      return callback(dashboard);
+    for (var key in appMap) {
+      var selected = key === initial;
+      dashboard.templating.list[0].options.push({text: key, value: key, selected: selected});
     }
 
+    dashboard.templating.list[0].options.sort(function(a, b) {
```
Comment (peggyl) [inline dashboards/scripted_services.js]: Why doesn't `dashboard.templating.list[0].options.sort()` work here? Isn't it just a lexicographic sort on env--app strings?

File: dashboards/scripted_services.js [hunk 4]
```diff
@@ -79,38 +110,65 @@ return function(callback) {
...
+    });
+
+    // Grafana only allows you to use the string value of the template params, you can't do anything
+    // else with it, such as use the ELB associated with the service to look up ELB stats.
+    // To get around this, we extract the currently selected env-app pair, monitor for changes, and
+    // reload the page with new query params if we get them.
+    var intervalID;
+    intervalID = setInterval(function() {
+      var current = $("a.variable-value-link.ng-binding").text().trim();
+      if (current == initial) {
```
Comment (peggyl) [inline dashboards/scripted_services.js]: `===`?

**Comment by peggyl** [comment]
Does it make sense (maybe in a separate PR) to cache the service mappings in browser local storage or something? The user-visible performance impact isn't awful, but it would be nice to save unnecessary work if possible.

File: dashboards/scripted_services.js (hunk 3 above)
Comment (azylman) [inline dashboards/scripted_services.js]: No, `options` is a list of objects with key-val pairs

**Pull request merged**